helixbass/tree-sitter-lint
# Basic plugin loading

In this PR:
- allow specifying a `plugins` section of the config file and configuring rules from that plugin in the `rules` section
- basic "API" for what a plugin crate is expected to expose
- update the hard-coded "custom local binary" project to include an example plugin

To test:
Per previous PR testing instructions first manually re-build the local custom binary
If you add the identifier `foo` somewhere in the repo and do `cargo run` you should see it flagged with the rule name in the violation output showing up as `replace-foo-with-bar/replace-foo-with-bar`

Based on `config-file`
